File: backend/src/api_schema/individual_values.go
package api_schema

type IndividualValues struct {
	Id             string `json:"id"`
	HitPoints      int    `json:"hit_points"`
	Attack         int    `json:"attack"`
	Defense        int    `json:"defense"`
	SpecialAttack  int    `json:"special_attack"`
	SpecialDefense int    `json:"special_defense"`
	Speed          int    `json:"speed"`
}

File: frontend/src/entity/IndividualValues.ts
export interface IndividualValues {
    id: string;
    hitPoints: number;
    attack: number;
    defense: number;
    specialAttack: number;
    specialDefense: number;
    speed: number;
}

export const NewIndividualValuesFromJson = (individualValues: any): IndividualValues => {
    return {
        id: individualValues.id,
        hitPoints: individualValues.hit_points,
        attack: individualValues.attack,
        defense: individualValues.defense,
        specialAttack: individualValues.special_attack,
        specialDefense: individualValues.special_defense,
        speed: individualValues.speed,
    };
}